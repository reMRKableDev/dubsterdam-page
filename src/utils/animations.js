import { keyframes } from "styled-components";
import { bounce, fadeInUp, fadeInDown, shake, merge } from "react-animations";

export const bounceAnimation = keyframes`${bounce}`;
export const fadeInUpAnimation = keyframes`${fadeInUp}`;
export const fadeInDownAnimation = keyframes`${fadeInDown}`;
export const shakeAnimation = keyframes`${shake}`;
export const fadeInUpShakeAnimation = keyframes`${merge(fadeInUp, shake)}`;
