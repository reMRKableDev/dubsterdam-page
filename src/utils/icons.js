import { Mail, Github, Linkedin, Database } from "@styled-icons/feather";
import {
  Css3,
  Html5,
  ReactLogo,
  Nodejs,
  Javascript,
} from "@styled-icons/boxicons-logos";

export const mailIcon = Mail;
export const gitHubIcon = Github;
export const linkedInIcon = Linkedin;
export const databaseIcon = Database;
export const cssIcon = Css3;
export const htmlIcon = Html5;
export const reactIcon = ReactLogo;
export const nodeIcon = Nodejs;
export const javascriptIcon = Javascript;
