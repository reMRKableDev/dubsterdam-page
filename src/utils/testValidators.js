export const validateNotEmpty = (received) => {
  expect(received).not.toBeNull();
  expect(received).toBeTruthy();
};

export const validateDarkThemeMode = (received) => {
  expect(received).not.toMatch(/light/i);
  expect(received).toMatch(/dark/i);
};

export const validateLightThemeMode = (received) => {
  expect(received).not.toMatch(/dark/i);
  expect(received).toMatch(/light/i);
};
