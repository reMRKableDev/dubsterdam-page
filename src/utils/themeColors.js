import theme from "styled-theming";

export const bgColor = theme("mode", {
  light: "#fafafa",
  dark: "#222",
});

export const footerColor = theme("mode", {
  light: "#222",
  dark: "#fafafa",
});

export const textColor = theme("mode", {
  light: "#000",
  dark: "#fff",
});

export const buttonBgColor = theme("mode", {
  light: "#222",
  dark: "#eee",
});

export const buttonTextColor = theme("mode", {
  light: "#eee",
  dark: "#222",
});

export const brandColor = theme("mode", {
  light: "#ed6c6d",
  dark: "#ed6c6d",
});

export const textAccentColor = theme("mode", {
  light: "#555",
  dark: "#fff",
});
