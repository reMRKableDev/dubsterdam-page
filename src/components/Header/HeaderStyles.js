import styled from "styled-components";
import { bgColor, brandColor } from "../../utils/themeColors";
import { respondTo } from "../../layout/respondTo";
import { fadeInUpAnimation, shakeAnimation } from "../../utils/animations";

export const StyledHeader = styled.header`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

export const StyledFigure = styled.figure`
  animation: 1.2s ${shakeAnimation};
  animation-delay: 1.4s;
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;

  img {
    width: 30vw;
    height: auto;
  }
`;

export const StyledH1 = styled.h1`
  animation: 1.5s ${fadeInUpAnimation};
  font-size: 8vw;
  font-family: "Rubik", sans-serif;
  margin: 1em 0;
  text-transform: uppercase;

  span {
    color: ${brandColor};
    border: 4px solid ${brandColor};
    box-shadow: 0 0 10px ${!bgColor};
    padding: 5px;
    text-transform: none;
  }

  ${respondTo.sm`
    margin: 0.8em 0;

    span {
      border: 6px solid ${brandColor};
    }
  `}

  ${respondTo.md`
  margin: 0.6em 0;

  span {
    border: 8px solid ${brandColor};
  }
  `}
  
  ${respondTo.lg`
  margin: 0.5em 0;

  span {
    border: 10px solid ${brandColor};
  }
	`}
`;
