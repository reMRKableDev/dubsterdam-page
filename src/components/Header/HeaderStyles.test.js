import React from "react";
import { shallow } from "enzyme";
import { StyledHeader, StyledFigure, StyledH1 } from "./HeaderStyles";

import { validateMatchSnapshot } from "../../utils/testValidators";

describe("Header Styles unit tests", () => {
  it("should validate match to snapshot of StyledHeader", () => {
    const wrapper = shallow(<StyledHeader />);
    validateMatchSnapshot(wrapper);
  });

  it("should validate match to snapshot of StyledFigure ", () => {
    const wrapper = shallow(<StyledFigure />);
    validateMatchSnapshot(wrapper);
  });

  it("should validate match to snapshot of StyledH1 ", () => {
    const wrapper = shallow(<StyledH1 />);
    validateMatchSnapshot(wrapper);
  });
});
