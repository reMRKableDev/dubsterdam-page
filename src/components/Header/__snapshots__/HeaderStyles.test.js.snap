// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Styles unit tests should validate match to snapshot of StyledFigure  1`] = `
.c0 {
  -webkit-animation: 1.2s fzrbgR;
  animation: 1.2s fzrbgR;
  -webkit-animation-delay: 1.4s;
  animation-delay: 1.4s;
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;
}

.c0 img {
  width: 30vw;
  height: auto;
}

<figure
  className="c0"
/>
`;

exports[`Header Styles unit tests should validate match to snapshot of StyledH1  1`] = `
.c0 {
  -webkit-animation: 1.5s dMrtAp;
  animation: 1.5s dMrtAp;
  font-size: 8vw;
  font-family: "Rubik",sans-serif;
  margin: 1em 0;
  text-transform: uppercase;
}

.c0 span {
  border: 4px solid;
  box-shadow: 0 0 10px;
  padding: 5px;
  text-transform: none;
}

@media (min-width:768px) {
  .c0 {
    margin: 0.8em 0;
  }

  .c0 span {
    border: 6px solid;
  }
}

@media (min-width:992px) {
  .c0 {
    margin: 0.6em 0;
  }

  .c0 span {
    border: 8px solid;
  }
}

@media (min-width:1200px) {
  .c0 {
    margin: 0.5em 0;
  }

  .c0 span {
    border: 10px solid;
  }
}

<h1
  className="c0"
/>
`;

exports[`Header Styles unit tests should validate match to snapshot of StyledHeader 1`] = `
.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

<header
  className="c0"
/>
`;
